### ISSUE TITLE
mpConfig: run as TCK as FAT: @Inject Config: property "env.sun.java.command" does not match System.getenv() value

### ISSUE THREAD

**Issue by hutchig** [OpenLiberty/open-liberty#1435]
ConfigProviderTest: 
```
    @Test
    public void testEnvironmentConfigSource() {
        Map<String, String> env = System.getenv();
        for (Map.Entry<String, String> envEntry : env.entrySet()) {
            Assert.assertEquals(envEntry.getValue(), config.getValue(envEntry.getKey(), String.class));
        }
    }
```
Fails with: 

 expected  [/libertyGit/open-liberty/dev/build.image/wlp/bin/tools/ws-server.jar  FATServer --clean] but found  [org.gradle.launcher.daemon.bootstrap.GradleDaemon 4.3.1]
for env variable "env.sun.java.command"

System.getenv() gives
          <property name="env.sun.java.command" value="org.gradle.launcher.daemon.bootstrap.GradleDaemon 4.3.1" />

The @Inject Config has the property: 
name=env.sun.java.command value=
"/libertyGit/open-liberty/dev/build.image/wlp/bin/tools/ws-server.jar FATServer --clean"

This occurs when the test is run as an Arquillian based FAT test.

The GradleDaemon looks like it is related to the "./gradlew com.ibm.ws.microprofile.config_fat_tck:buildandrun" that was used to
launch the FAT bucket. Whereas the ws-server.jar looks like it was the
command that was used to start the server. 

/libertyGit/open-liberty/dev/com.ibm.ws.microprofile.config_fat_tck/build/libs/autoFVT/results/junit/TESTS-TestSuites.xml can be used to see all the environment variables. 

I do not remember having a problem with this (old) test when we ran
the TCK using a tckRunner project and "mvn test -DsuiteXmlFile=tck-suite.xml"
from the command line so it is as if the Arquillian environment 

There is no special error or debug data to record here.

**Comment by hutchig** [comment]
tools/ws-server.jar versus server.....
[01/24/2018 22:59:48:271 GMT] 013 LocalProvider                  shArrayTransform               I Transformed Command now is: /libertyGit/open-liberty/dev/build.image/wlp/bin/server start FATServer --clean

**Comment by hutchig** [comment]
Tried manipulating these values from arquillian, surefire, server.env, bootstrap.properties, the ProcessBuilder in the FAT test - to no effect.

**Comment by hutchig** [comment]
Also see output.txt:[01/25/2018 05:53:08:491 GMT] 013 LocalProvider                  shArrayTransform               I Transformed Command now is: /libertyGit/open-liberty/dev/build.image/wlp/bin/server start FATServer --clean

**Comment by tevans78** [comment]
We now know that the test is bad. It will be fixed in 1.2.1 of the tck and when that is released, we can re-enable the test for config 1.2.
